holmes/core/conversations_worker/realtime_manager.py

```
@@ -0,0 +1,539 @@
+
+        # skip proxy for localhost targets
+        if parsed.hostname in ("localhost", "127.0.0.1"):
+            return await ws_connect(url, *args, **kwargs)
+
+        port = parsed.port or (443 if parsed.scheme == "wss" else 80)
+        proxy = _SocksProxy.from_url(proxy_connect_url)
+        sock = await proxy.connect(dest_host=parsed.hostname, dest_port=port)
+        kwargs.setdefault("server_hostname", parsed.hostname)
+        if parsed.scheme == "wss" and "ssl" not in kwargs:
+            kwargs["ssl"] = ssl.create_default_context()
+        return await ws_connect(url, sock=sock, *args, **kwargs)
```

Ruff (0.15.9)

[warning] Star-arg unpacking after a keyword argument is strongly discouraged

(B026)